js/runner_test.go

```diff
@@ -2001,3 +2001,98 @@ func TestComplicatedFileImportsForGRPC(t *testing.T) {
 		})
 	}
 }
+
+func TestExecutionStats(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name, script, expErr string
+	}{
+		{"vu_ok", `
+		var exec = require('k6/execution');
+
+		exports.default = function() {
+			var vuStats = exec.getVUStats();
+			if (vuStats.id !== 1) throw new Error('unexpected VU ID: '+vuStats.id);
+			if (vuStats.iteration !== 1) throw new Error('unexpected VU iteration: '+vuStats.iteration);
+		}`, ""},
+		{"vu_err", `
+		var exec = require('k6/execution');
+		exec.getVUStats();
+		`, "VU information can only be returned from an exported function"},
+		{"scenario_ok", `
+		var exec = require('k6/execution');
+		var sleep = require('k6').sleep;
+
+		exports.default = function() {
+			var ss = exec.getScenarioStats();
+			sleep(0.1);
+			// goja's Date handling is weird, see https://github.com/dop251/goja/issues/170
+			var startTime = new Date(JSON.parse(JSON.stringify(ss.startTime)));
+			if (ss.name !== 'default') throw new Error('unexpected scenario name: '+ss.name);
+			if (ss.executor !== 'test-exec') throw new Error('unexpected executor: '+ss.name);
+			if (startTime > new Date()) throw new Error('unexpected startTime: '+startTime);
+			if (ss.progress !== 0.1) throw new Error('unexpected progress: '+ss.progress);
+		}`, ""},
+		{"scenario_err", `
+		var exec = require('k6/execution');
+		exec.getScenarioStats();
+		`, "scenario information can only be returned from an exported function"},
+		{"test_ok", `
+		var exec = require('k6/execution');
+
+		exports.default = function() {
+			var ts = exec.getTestStats();
+			if (ts.vus_active !== 1) throw new Error('unexpected vus_active: '+ts.vus_active);
+			if (ts.vus_max !== 0) throw new Error('unexpected vus_max: '+ts.vus_max);
+			if (ts.iterations_completed !== 0) throw new Error('unexpected iterations_completed: '+ts.iterations_completed);
+			if (ts.iterations_interrupted !== 0) throw new Error('unexpected iterations_interrupted: '+ts.iterations_interrupted);
```

Comment:
These checks are kind of pointless... Should I move these tests to some higher-level ones that involve actual executors?